modules/network/traffic-manager/main.bicep

```diff
@@ -50,6 +50,58 @@ param monitorConfig object = {
   ]
 }
 
+@description('Specifies the number of days that logs will be kept for; a value of 0 will retain data indefinitely. default is 365.')
+@minValue(0)
+@maxValue(365)
+param diagnosticLogsRetentionInDays int = 365
+
+@description('Resource ID of the diagnostic storage account.')
+param diagnosticStorageAccountId string = ''
+
+@description('Resource ID of the diagnostic log analytics workspace.')
+param diagnosticWorkspaceId string = ''
+
+@description('Resource ID of the diagnostic event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to.')
+param diagnosticEventHubAuthorizationRuleId string = ''
+
+@description('Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category.')
+param diagnosticEventHubName string = ''
+
+@description('The name of logs that will be streamed. default is allLogs.')
+@allowed([
+  'allLogs'
+  'ProbeHealthStatusEvents'
+])
+param logsToEnable string = 'allLogs'
+
+@description('The name of metrics that will be streamed. default is AllMetrics')
+@allowed([
+  'AllMetrics'
+])
```

Comment:
```suggestion
```
There are lots of other metrics allowed, don't want to restrict the user to only this value.

Reply:
but on portal for TM, only shows AllMetrics, do we still wanted to remove allowed?

Reply:
@dciborow what your opinion on above ?

Reply:
I havent done to much digging here... If its the ONLY allowed value, then we should make it a var instead of a param. 

But my knee-jerk reaction is still to just remove the allowed... because my gut says there are more values for this. (I put a link to the potential values for a similiar use case somewhere, let me try and find that...)

Reply:
okay, so per: https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings?tabs=portal#metric-category-isnt-supported

lets move this to a variable, since it looks like it should always be 'AllMetrics'

Reply:
done!